### PR TITLE
Feat/update to most recent

### PR DESCRIPTION
**Pull request by ZenGround0**
@hsanjuan I addressed the changes you brought up
@Kubuxu could you take a look too?

Thanks!

### CODE REVIEW

**Comment by Kubuxu** [comment]
@ZenGround0 could you explain the reason behind changes?
It is a bit hard to review code when it is in a different file (GH UI problem, git handles it I think) and when you don't have cause for the changes.

**Comment by ZenGround0** [comment]
@Kubuxu This should be easier to review.  Let me know if you need anything else from me.  I think @frrist made most of these changes so he can answer questions about specific refactoring choices too.

File: fskeystore.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"os"
 	"path/filepath"
 
-	ci "github.com/libp2p/go-libp2p-crypto"
```
Comment (Kubuxu) [inline fskeystore.go]: gx paths in small deps are not needed, we use it mostly in go-ipfs to enforce the versions.

File: keystore.go [hunk 2]
```diff
@@ -2,13 +2,9 @@ package keystore
...
 	"fmt"
-	"io/ioutil"
-	"os"
-	"path/filepath"
-	"strings"
 
-	logging "github.com/ipfs/go-log"
-	ci "github.com/libp2p/go-libp2p-crypto"
+	logging "gx/ipfs/QmRb5jh8z2E8hMGN2tkvs1yHynUanqnZ3UeKwgN1i9P1F8/go-log"
+	ci "gx/ipfs/QmaPbCnUMBohSGo3KnxEa2bHqyJVVeEEcwtqJAYxerieBo/go-libp2p-crypto"
```
Comment (Kubuxu) [inline keystore.go]: gx paths in small deps are not needed, we use it mostly in go-ipfs to enforce the versions.

Just un-rewrite it and it think it is good to go.

**Comment by ZenGround0** [comment]
@Kubuxu turned out I can get away with non-gx import paths.  Thanks for the review.

**Pull request merged**